plugins/inputs/intel_powerstat/file.go

```diff
@@ -0,0 +1,154 @@
+// +build linux
+
+package intel_powerstat
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// fileService is responsible for handling operations on files.
+type fileService interface {
+	getCPUInfoStats() (map[string]*cpuInfo, error)
+	getStringsMatchingPatternOnPath(path string) ([]string, error)
+	readFile(path string) ([]byte, error)
+	readFileToFloat64(reader io.Reader) (float64, int64, error)
+	readFileAtOffsetToUint64(reader io.ReaderAt, offset int64) (uint64, error)
+}
+
+type fileServiceImpl struct {
+}
+
+// getCPUInfoStats retrieves basic information about CPU from /proc/cpuinfo.
+func (fs *fileServiceImpl) getCPUInfoStats() (map[string]*cpuInfo, error) {
+	path := "/proc/cpuinfo"
+	cpuInfoFile, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading %s, err: %v", path, err)
+	}
+	defer cpuInfoFile.Close()
+
+	scanner := bufio.NewScanner(cpuInfoFile)
+
+	processorRegexp := regexp.MustCompile(`^processor\t+:\s([0-9]+)\n*$`)
+	physicalIDRegexp := regexp.MustCompile(`^physical id\t+:\s([0-9]+)\n*$`)
+	coreIDRegexp := regexp.MustCompile(`^core id\t+:\s([0-9]+)\n*$`)
+	vendorIDRegexp := regexp.MustCompile(`^vendor_id\t+:\s([a-zA-Z]+)\n*$`)
+	cpuFamilyRegexp := regexp.MustCompile(`^cpu\sfamily\t+:\s([0-9]+)\n*$`)
+	modelRegexp := regexp.MustCompile(`^model\t+:\s([0-9]+)\n*$`)
+	flagsRegexp := regexp.MustCompile(`^flags\t+:\s(.+)\n*$`)
+
+	stats := make(map[string]*cpuInfo)
+	currentInfo := &cpuInfo{}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		processorRes := processorRegexp.FindStringSubmatch(line)
+		if len(processorRes) > 1 {
+			currentInfo = &cpuInfo{
+				cpuID: processorRes[1],
+			}
```

Comment:
Shouldn't this be 
```suggestion
			currentInfo.cpuID = processorRes[1]
```

Reply:
Agreed that this needs to created for every new logical cpu found in `/proc/cpuinfo`